spec/query_associations_spec.cr

```diff
@@ -38,7 +47,9 @@ describe "Query associations" do
       .post_id(post_without_matching_comment.id)
       .create
 
-    posts = Post::BaseQuery.new.inner_join_comments.where_comments(&.body.eq("matching"))
+    posts = Post::BaseQuery.new
+      .inner_join_comments
+      .where_comments(Comment::BaseQuery.new.body.eq("matching"), join: false)
```

Comment:
This feels confusing to me. First we join comments then we `join: false`? I'm not sure what this is doing.

Reply:
We made it so  the where automatically does an inner join. If you want a different join  you'd  want  to disable that behavior. Maybe there is some other better way to name it?

`auto_join: false` or maybe `skip_join: true`?

Reply:
Do you have an easy way to paste what this whole statement is doing in SQL? Maybe that would help me understand what it's doing. I'm not getting how we're joining comments in the query, then later doing a `WHERE` and then specifying joining behavior again (which doesn't seem like it should have anything to do with `WHERE`).

Reply:
`where_comments(SomeQuery, automatically_add_join: false)`? Maybe not 100% self explanatory, but easier to google I think

Reply:
```
SELECT posts.custom_id, posts.created_at, posts.updated_at, posts.title, posts.published_at FROM posts INNER JOIN comments ON posts.custom_id = comments.post_id WHERE comments.body = $1
```

But if you did left_join_comments it will do a left_join instead

Reply:
Okay, I checked out the branch and am playing around with it and am still confused. All four of these post the exact same SQL:

```crystal
Post::BaseQuery.new
  .where_comments(Comment::BaseQuery.new.body.eq("matching"))

Post::BaseQuery.new
  .inner_join_comments
  .where_comments(Comment::BaseQuery.new.body.eq("matching"))

Post::BaseQuery.new
  .inner_join_comments
  .where_comments(Comment::BaseQuery.new.body.eq("matching"), join: false)

Post::BaseQuery.new
  .where_comments(Comment::BaseQuery.new.body.eq("matching"), join: true
```

So I'm confused why we use both `inner_join_{{assoc}}` _and_ `join: {{Boolean}}`. I think I'm missing something deadly obvious 😅. Or maybe it's besides the point in this branch?

Reply:
Ok so part of this is that in SQL a join and inner join are the same  https://softwareengineering.stackexchange.com/questions/206035/join-vs-inner-join-and-full-outer-join

We have both though in case you want to clarify it. Maybe we should remove the inner join method though. It probably just adds to the confusion. 

So let’s ignore the inner join thing for now. The main reason you’d use “join: false” is if you’re doing a left, right, or full join (or some other custom join) that you want to use. In that case you don’t want Avram to add an additional inner join so you use “join:false” or whatever we decide to call that option

Reply:
Ah ha! So where_{{assoc}} does an automatic _inner_ join for you and you can tell it _not_ to do that because you want to do some other kind of join with `join: false`. Is that right?

Reply:
Would it be a reach for this PR to instead do `join: INNER` or `join: NONE` or `join: LEFT` or something?

Reply:
Yep you nailed it!

I think that the problem there is that now we’d introduce more ways to join stuff. You can use the join methods or this other kind of option. It’s also be trickier to make type safe as well

I think we definitely need a better name for this option though. Maybe “add_join: false” or automatically_add_join: false?

A bit long but I think this won’t be used much so clarity is probably more important than keeping it super short. 

Thoughts @jwoertink?

Reply:
For the record, I think what threw me into this confusion was the spec. It has the `inner_join_comments` and `join: false`. I think if it hadn't had the `inner_join_comments` I would have been much less confused.

`auto_join: false`?

Reply:
Did the left_join and  right_join ones  not  confuse you? https://github.com/luckyframework/avram/pull/172/files#diff-54c72b6e1799559414b793e3fa88bc3eR73

Just  wondering if  it  was specifically the inner_join (because I think we should remove it to reduce confusion)

Reply:
And I think `auto_join` works!

Reply:
Yeah, I think they are confusing also. It's the join twice that's weird. It might make me think that I'm canceling the left_join

Reply:
`auto_inner_join: false` might be even better if we can stand the length.

Reply:
Yeah I can definitely see that.  Like `join: false` is removing the join completely or something. `auto_inner_join` I *love*. Nice and  clear  that it is  one specific type of join we are not doing!

Reply:
Yay, we did it! ✋